core/float/shared/arithmetic_exception_in_coerce.rb

```diff
@@ -0,0 +1,36 @@
+require File.expand_path('../../fixtures/classes', __FILE__)
+
+describe :float_arithmetic_exception_in_coerce, shared: true do
+  ruby_version_is ""..."2.5" do
+    it "rescues exception (StandardError and subclasses) raised in other#coerce and raises TypeError" do
+      b = mock("numeric with failed #coerce")
+      b.should_receive(:coerce).and_raise(FloatSpecs::CoerceError)
+
+      # e.g. 1.0 > b
+      -> { 1.0.send(@method, b) }.should raise_error(TypeError, /MockObject can't be coerced into Float/)
+    end
+
+    it "does not rescue Exception and StandardError siblings raised in other#coerce" do
+      [ Exception.new, NoMemoryError.new, ScriptError.new, SecurityError.new,
+        SignalException.new('INT'), SystemExit.new, SystemStackError.new
```

Comment:
Seems fairly exhaustive :) I think 1 or 2 examples would be enough here, as we don't really want to test all the relations between these exception classes in this spec. Could you reduce it to 1 or 2 example classes?